### PR TITLE
Add completions.sort option

### PR DESCRIPTION
**Pull request by ysthakur**
# Description

More or less closes https://github.com/nushell/nushell/issues/13295, although I didn't mess with the help completer. This PR gets rid of the `SortBy` enum so that individual completers can no longer choose how they want to sort completions. Instead, a new `sort` field is added to `CompletionOptions` (configurable through `$env.config.completions.sort`). If set to `"default"` (the default), completions will be sorted based on match algorithm (alphabetical order for prefix matching, fuzzy matcher score for fuzzy matching). If set to `"alphabetical"`, completions will be sorted in alphabetical order.

This is my first time adding a new config option, so let me know if I've missed anything.

# User-Facing Changes

There's a new `$env.config.completions.sort` option to control sort order. Users will now see all their completions sorted either in alphabetical order or based on the fuzzy matcher score, based on their chosen match algorithm. Previously, some completers would've sorted in alphabetical order and some by levenshtein distance.

Now, levenshtein distance isn't used at all. Note that the fuzzy matcher score will not provide the exact same results as levenshtein distance. I personally think the fuzzy matcher score is a better metric than levenshtein distance (it's easier for me to get the completions I want), but if anyone likes levenshtein distance, we can figure out a way to keep that too.

Additionally, with custom completions, if a provided completion has `sort: true`, then the suggestions provided with it will be sorted in alphabetical order. Otherwise, they'll be sorted based on the user's chosen match algorithm. It would probably have made more sense to use `sort: "alphabetical"/"default"` here too, but I didn't want to completely break anything. Let me know what behavior would make more sense here.

# Tests + Formatting

- Added one test that uses `algorithm = "fuzzy"` with `sort = "alpha"`.
- Changed the LSP's keyword completion test to complete `overlay`. Previously, it completed `def` (from `de`), but now, `debug` and its subcommands are provided as completions before `def`, because they come before it in alphabetical order, even if previously, their levenshtein distance to `de` was greater.

# After Submitting

This PR recomputes fuzzy matcher scores when suggestions are to be sorted (they're already computed once when filtering the list of suggestions). It's possible to compute those scores only once, but that requires more extensive changes. I've got a PR for that lined up; I can open it if this is merged.

### CODE REVIEW

File: crates/nu-cli/src/completions/completion_common.rs [hunk 1]
```diff
@@ -273,33 +274,32 @@ pub fn adjust_if_intermediate(
...
     items: Vec<SemanticSuggestion>,
-    sort_by: SortBy,
+    options: &CompletionOptions,
 ) -> Vec<SemanticSuggestion> {
-    sort_completions(prefix, items, sort_by, |it| &it.suggestion.value)
+    sort_completions(prefix, items, options, |it| &it.suggestion.value)
 }
 
 /// # Arguments
 /// * `prefix` - What the user's typed, for sorting by Levenshtein distance
```
Comment (fdncred) [inline crates/nu-cli/src/completions/completion_common.rs:283]: Is Levenshtein still a thing?

**Comment by fdncred** [comment]
It looks like LevenshteinDistance is gone, but I could be reading it wrong. I'm ok with that but I'm not sure others will be.

File: crates/nu-cli/src/completions/completion_common.rs (hunk 1 above)
Comment (ysthakur) [reply crates/nu-cli/src/completions/completion_common.rs]: Good catch. Yeah, I just missed that comment, Levenshtein distance is gone now.

**Comment by kubouch** [comment]
I'd rename "alpha" to "alphabetical". Ať first, I thought it's some coefficient Greek letter alpha. Otherwise I'm fine with it.

**Comment by ysthakur** [comment]
@kubouch Done

File: crates/nu-utils/src/sample_config/default_config.nu [hunk 2]
```diff
@@ -205,6 +205,7 @@ $env.config = {
         quick: true    # set this to false to prevent auto-selecting completions when only one remains
         partial: true    # set this to false to prevent partial filling of the prompt
         algorithm: "prefix"    # prefix or fuzzy
+        sort: "default" # "default" (choose based on algorithm) or "alphabetical"
```
Comment (fdncred) [inline crates/nu-utils/src/sample_config/default_config.nu:208]: When a user sees this i'm not sure they'll know the difference between default and alphabetical. I'm wondering if there's something better than "default" that can spell out what it does better.

**Comment by fdncred** [comment]
If you can fix the conflicts, and tweak my last request we can probably land this and start using it.

File: crates/nu-utils/src/sample_config/default_config.nu (hunk 2 above)
Comment (ysthakur) [reply crates/nu-utils/src/sample_config/default_config.nu]: How about "smart"? (sorry, been away for a while)

Comment (fdncred) [reply crates/nu-utils/src/sample_config/default_config.nu]: ya, smart seems better. good choice.

**Comment by ysthakur** [comment]
Alright, workflows are green again, shall I merge?

**Pull request merged**